internal/distributedtracing/distributedtracing.go

```diff
@@ -167,12 +162,7 @@ func parseDistributedTracingConfig(raw []byte) (*distributedTracingConfig, error
 	return &config, nil
 }
 
-func (c *distributedTracingConfig) validateAndInjectDefaults() error {
```

Comment:
This should be changed back to include an error return, and validate that `c.Type` is either empty-string or "grpc", return an error for anything else.